src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -1,109 +1,165 @@
 <?php
+
 namespace CubeTools\CubeCommonBundle\DataHandling\Logs;
 
 use DataDog\AuditBundle\Entity\AuditLog;
 use DataDog\AuditBundle\Entity\Association;
 use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ORM\QueryBuilder;
 
 /**
  * Class handling data from DataDogAuditBundle.
  */
 class DataDogAudit implements LogsInterface
 {
-    /**
-     * key for subtable with deleted elements
-     */
-    const KEY_REMOVE = 'removed';
+    use LogsFunctionsTrait;
+
+    const TEMP_KEY_READD = 'temp_readd';
+    const TEMP_KEY_OLDVAL = 'temp_oldval';
 
     /**
-     * key for subtable with added elements
+     * @var ObjectManager
      */
-    const KEY_ADD = 'added';
+    protected $em;
 
     /**
-     * key for subtable with elements, which has not changed
+     * @var mixed[] cached data for current entity type
      */
-    const KEY_UNCHANGED = 'unchanged';
+    protected $cache;
 
     /**
-     * @var \Doctrine\Common\Persistence\ObjectManager
+     * @var mixed[] cached data for current entity
      */
-    protected $em;
+    protected $instanceCache = array();
 
     public function __construct(ObjectManager $em)
     {
         $this->em = $em;
     }
 
     /**
-     * Method for getting all versions of given entity.
-     * @param Object $entity entity for which we want to get all versions
-     * @return array versions of entity
+     * Method for getting AuditLog of all versions of given entity.
+     *
```

Comment:
is this additional line necessary?

Reply:
Yes, this separates the 1-line-summary from more details.